src/main/java/Sb2.java

```diff
@@ -23,24 +25,57 @@ public Sb2(String filePath) throws IOException {
 
         String pathToPackageJson = destPath + File.separator + "package.json";
         String jsonString = getFileContents(pathToPackageJson);
-        sb2Json = createJSONObject(jsonString);
+        init(createJSONObject(jsonString));
+    }
+    /**
+     * Construct an Sb2 object from a JSONObject. Useful for testing.
+     * @param stage JSONObject from which to construct Sb2
+     */
+    public Sb2(JSONObject stage) {
+        init(stage);
     }
     /**
+     * Function to be called from all constructors.
+     * @param stage JSONObject which is underlying data structure for Sb2.
+     */
+    public void init(JSONObject stage) {
+        this.stage = stage;
+        this.sprites = new Sprites(stage);
+    }
+
+    /**
      * Return underlying JSONObject.
      * @return The underlying JSONObject.
      */
     public JSONObject getJSONObject() {
-        return sb2Json;
+        return stage;
     }
 
     /**
-     * Return the number of Sprites in Sb2 object.
-     * @return The number of sprites in project
+     * Each sprite has a unique name.
+     * This method returns these names in a array of strings.
+     * @return Array of sprite names.
      */
-    public int countSprites() {
-        return 0;
+    public String[] getSpriteNames() {
```

Comment:
@whereswaldon, I knew I'd seen that pattern before.  It's called object composition.  See page 3 of http://www.artima.com/designtechniques/compoinh.html
There really isn't any reason to use it here, however.  Fenwick said we needed to implement those methods, but he didn't say where.  I don't think that Sb2 really needs to have those methods itself.  If it did, however, this pattern would be a fine way (imho) to get do it and still make the functionality of Sprites modular and so reusable and more maintainable.